agent/pkg/k8s/certificates.go

```diff
@@ -0,0 +1,198 @@
+package k8s
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	pb "github.com/buoyantio/linkerd-buoyant/gen/bcloud"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	controlPlaneComponentLabel   = "linkerd.io/control-plane-component"
+	identityComponentName        = "identity"
+	linkerdProxyContainerName    = "linkerd-proxy"
+	linkerdRootsEnvVarName       = "LINKERD2_PROXY_IDENTITY_TRUST_ANCHORS"
+	proxyAdminPortName           = "linkerd-admin"
+	linkerdNsEnvVarName          = "_l5d_ns"
+	linkerdTrustDomainEnvVarName = "_l5d_trustdomain"
+)
+
+func (c *Client) GetControlPlaneCerts() (*pb.ControlPlaneCerts, error) {
```

Comment:
Btw, I think, a good way to test that end to end is to supply a pod in the fake k8s API with a container IP that is localhost with some port, and spin a fake TLS server locally that will return the expected issuer certs when connected to. I actually plan to attempt that once I get some initial feedback.